pkg/gui/commit_message_panel.go

```diff
@@ -51,6 +51,89 @@ func (gui *Gui) handleCommitFocused(g *gocui.Gui, v *gocui.View) error {
 	return gui.renderString(g, "options", message)
 }
 
+type credentials chan string
+
+// waitForPassUname wait for a username or password input from the pushPassUname popup
+func (gui *Gui) waitForPassUname(g *gocui.Gui, currentView *gocui.View, passOrUname string) string {
+	gui.credentials = make(chan string)
+	pushPassUnameView, _ := g.View("pushPassUname")
+	if passOrUname == "username" {
+		pushPassUnameView.Title = gui.Tr.SLocalize("PushUsername")
+		pushPassUnameView.Mask = 0
+	} else {
+		pushPassUnameView.Title = gui.Tr.SLocalize("PushPassword")
+		pushPassUnameView.Mask = '*'
+	}
+	g.Update(func(g *gocui.Gui) error {
+		_, err := g.SetViewOnTop("pushPassUname")
```

Comment:
While testing i have never had a crash because of deleting that view.  
I have tried switching it to `.SetViewOnBottom` but then the view doesn't go away, instead the application hangs with a password input box

![image](https://user-images.githubusercontent.com/15320763/49718094-5bf6f400-fc59-11e8-8d1d-ccba93b43630.png)